src/client.rs

```diff
@@ -19,8 +21,41 @@ use tokio_service::Service;
 type WireResponse<Resp, E> = Result<Result<Resp, WireError<E>>, DeserializeError>;
 type ResponseFuture<Req, Resp, E> = futures::Map<<BindClient<Req, Resp, E> as Service>::Future,
                                             fn(WireResponse<Resp, E>) -> Result<Resp, ::Error<E>>>;
-type BindClient<Req, Resp, E> =
-    <Proto<Req, Result<Resp, WireError<E>>> as ProtoBindClient<Multiplex, TcpStream>>::BindClient;
+type BindClient<Req, Resp, E> = <Proto<Req, Result<Resp, WireError<E>>> as
+                                ProtoBindClient<Multiplex, StreamType>>::BindClient;
+
+/// TLS-specific functionality
+#[cfg(feature = "tls")]
+pub mod tls {
+    use native_tls::TlsConnector;
+
+    /// TLS context for client
+    pub struct Context {
+        /// Domain to connect to
+        pub domain: String,
+        /// TLS connector
+        pub tls_connector: TlsConnector,
+    }
+
+    impl Context {
+        /// Try to construct a new `Context`, providing the domain the client will
```

Comment:
Good point, actually this comment is wrong. It's the domain used for validation, not connection. I will update that.